src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -5,58 +5,72 @@
         /// <param name="dataSample">The optional location of a data sample.</param>
         public static TextLoader TextReader(this DataOperations catalog,
-            TextLoader.Arguments args, IMultiStreamSource dataSample = null)
-            => new TextLoader(CatalogUtils.GetEnvironment(catalog), args, dataSample);
+            Column[] columns, bool hasHeader = false, char separatorChar = '\t', IMultiStreamSource dataSample = null)
+            => new TextLoader(CatalogUtils.GetEnvironment(catalog), columns, hasHeader, separatorChar, dataSample);
 
         /// <summary>
-        /// Create a text reader.
+        /// Create a text reader <see cref="TextLoader"/>.
         /// </summary>
         /// <param name="catalog">The catalog.</param>
-        /// <param name="columns">The columns of the schema.</param>
-        /// <param name="advancedSettings">The delegate to set additional settings.</param>
-        /// <param name="dataSample">The optional location of a data sample.</param>
-        public static TextLoader TextReader(this DataOperations catalog,
-            TextLoader.Column[] columns, Action<TextLoader.Arguments> advancedSettings = null, IMultiStreamSource dataSample = null)
-            => new TextLoader(CatalogUtils.GetEnvironment(catalog), columns, advancedSettings, dataSample);
+        /// <param name="args">Defines the settings of the load operation.</param>
+        /// <param name="dataSample">Allows to expose items that can be used for reading.</param>
+        public static TextLoader TextReader(this DataOperations catalog, Arguments args, IMultiStreamSource dataSample = null)
+            => new TextLoader(CatalogUtils.GetEnvironment(catalog), args, dataSample);
 
         /// <summary>
         /// Read a data view from a text file using <see cref="TextLoader"/>.
         /// </summary>
         /// <param name="catalog">The catalog.</param>
         /// <param name="columns">The columns of the schema.</param>
-        /// <param name="advancedSettings">The delegate to set additional settings</param>
-        /// <param name="path">The path to the file</param>
+        /// <param name="hasHeader">Whether the file has a header.</param>
+        /// <param name="separatorChar"> The character used as separator between data points in a row. By default the tab character is used as separator.</param>
+        /// <param name="path">The path to the file.</param>
         /// <returns>The data view.</returns>
         public static IDataView ReadFromTextFile(this DataOperations catalog,
-            TextLoader.Column[] columns, string path, Action<TextLoader.Arguments> advancedSettings = null)
+            string path, Column[] columns, bool hasHeader = false, char separatorChar = '\t')
+        {
+            Contracts.CheckNonEmpty(path, nameof(path));
+
+            var env = catalog.GetEnvironment();
+
+            // REVIEW: it is almost always a mistake to have a 'trainable' text loader here.
```

Comment:
>// REVIEW: it is almost always a mistake to have a 'trainable' text loader here [](start = 12, length = 79)

Did it work well if you specify header to true and didn't pass dataSample?

Reply:
Yes, check this test         public void CustomTransformer()


---
In reply to: [239161201](https://github.com/dotnet/machinelearning/pull/1784#discussion_r239161201) [](ancestors = 239161201)